internal/rollup/terser-wrapped.js

```diff
@@ -74,51 +108,92 @@ function runterser(inputFile, outputFile, sourceMapFile) {
     },
     'mangle': !debug,
   };
+  return terserConfig;
+}
 
-  let config = configFile;
-  if (!config) {
-    config = tmp.fileSync({keep: false, postfix: '.json'}).name;
+/**
+ *
+ * @param {{inputFile: string, outputFile: string, sourceMapFile: string}[]} files
+ * @param {boolean} debug
+ */
+async function runTerserSerial(files, debug) {
+  const Terser = require(TERSER_MODULE_LOCATION);
+  for (const f of files) {
+    // run one set of terser at a time
+    await runTerser(f.inputFile, f.outputFile, f.sourceMapFile, Terser, debug);
   }
+}
 
-  fs.writeFileSync(config, JSON.stringify(terserConfig));
+/**
+ *
+ * @param {{inputFile: string, outputFile: string, sourceMapFile: string}[]} files
+ * @param {boolean} debug
+ */
+async function runTerserParallel(files, debug) {
+  // TODO: do some concurrency limiting here
+  await Promise.all(
+      files.map(f => runTerserThread(f.inputFile, f.outputFile, f.sourceMapFile, debug)));
+}
 
-  const args = [
-    require.resolve('build_bazel_rules_nodejs_rollup_deps/node_modules/terser/bin/uglifyjs'),
-    inputFile, '--output', outputFile, '--config-file', config
-  ];
+/**
+ *
+ * @param {string} inputFile
+ * @param {string} outputFile
+ * @param {string} sourceMapFile
+ * @param {boolean} debug
+ * @returns {Promise<void>}
+ */
+function runTerserThread(inputFile, outputFile, sourceMapFile, debug) {
+  return new Promise((resolve, reject) => {
+    const task = {
+      inputFile,
+      outputFile,
+      sourceMapFile,
+      terser: require.resolve(TERSER_MODULE_LOCATION),
+      debug
+    } const worker = new worker_threads.Worker(__filename, {workerData: task});
+
+    worker.on('exit', () => resolve());
+    worker.on('error', err => reject(err));
+  })
+}
 
-  for (arg in argv) {
-    const prefix = arg.length == 1 ? '-' : '--';
-    const value = argv[arg];
-    args.push(prefix + arg);
-    if (value && value !== true) {
-      args.push(value);
-    }
-  }
+/**
+ *
+ * @param {string} inputFile
+ * @param {string} outputFile
+ * @param {string} sourceMapFile
+ * @param {Function} Terser
+ * @param {boolean} debug
+ * @returns {Promise<void>}
+ */
+async function runTerser(inputFile, outputFile, sourceMapFile, Terser, debug) {
+  if (DEBUG) console.error(`Minifying ${inputFile} -> ${outputFile} (sourceMap ${sourceMapFile})`);
 
-  if (DEBUG) console.error(`Running node ${args.join(' ')}`);
+  const terserOptions = buildTerserConfig(sourceMapFile, debug)
+  const inputCode = (await fs.promises.readFile(inputFile)).toString();
```

Comment:
Are character encoding and BOM presence relevant here?

Reply:
I really don't know. We could add a `trim()` call which would remove the BOM if it is actually encoded as that.
I don't know the impact of stripping the BOM and re-writing it as plain UTF8 once we're done